jsapp/js/components/submissions/tableUtils.ts

```diff
@@ -272,3 +278,73 @@ export function getAllDataColumns(
 
   return output;
 }
+
+export interface TableFilterQuery {
+  queryString: string;
+  queryObj: {
+    [key: string]:
+      | string
+      | null
+      | {$in: string[]}
+      | {$regex: string; $options: string};
+  };
+}
+
+/**
+ * This function uses filters list from `react-table` output to produce queries
+ * that our Back end can understand. We use it it multiple places that intensely
+ * need to use identical output. We might simply return `queryObj` and make
+ * the code stringify it by itself, but it will make it less robust.
+ */
+export function buildFilterQuery(
+  /** Whole survey of given asset - we need it to get questions types */
+  survey: SurveyRow[],
+  /** List of `react-table` filters */
+  filters: Filter[]
+): TableFilterQuery {
+  const output: TableFilterQuery = {
+    queryString: '',
+    queryObj: {},
+  };
+
+  filters.forEach((filter) => {
+    switch (filter.id) {
+      case '_id': {
+        output.queryObj[filter.id] = {$in: [filter.value]};
+        break;
+      }
+      case VALIDATION_STATUS_ID_PROP: {
+        if (filter.value === VALIDATION_STATUSES.no_status.value) {
+          output.queryObj[filter.id] = null;
+        } else {
+          output.queryObj[filter.id] = filter.value;
+        }
+        break;
+      }
+      // Apart from few exceptions in above cases, we tend to treat all columns
+      // (`filter.id`s) with the same algorithm
+      default: {
+        // We assume `filter.id` is the question name (Data Table column name)
+        const foundRow = survey.find((row) => getRowName(row) === filter.id);
+
+        // Some question types needs the data to be filtered by exact values
+        // (e.g. "yes" shouldn't mach "yessica" or "yes and no")
+        if (foundRow && FILTER_EXACT_TYPES.includes(foundRow.type)) {
```

Comment:
The PR does fix the bug reported in #4745 but I'm kind of curious about this change here. Should we really be changing the default behaviour to match exact values (without any option)? I'd imagine this would be fairly different from what a user would be used to.

Reply:
Tino gave this change green light, see https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/Bulk.20update.20validation.20status.20bug/near/341412

There is also a spreadsheet that lists all question types and what behaviour we want filtering wise: https://docs.google.com/spreadsheets/u/1/d/1tVOeg7Sdu64qh22ku0KsGjs9t-Fl3g-DBVQa0htiRPQ/edit#gid=0